atomic_reactor/utils/cachito.py

```diff
@@ -182,7 +183,28 @@ def assemble_download_url(self, request):
         :return: str, the URL to download the sources
         """
         request_id = self._get_request_id(request)
-        return '{}/api/v1/requests/{}/download'.format(self.api_url, request_id)
+        return '{}/{}/download'.format(self.api_endpoint, request_id)
+
+    def get_icm(self, request):
+        """Get and validate the ICM ("Image Content Manifest") from the Cachito API
+
+        `content-manifest` endpoint.
+
+        :param request: int or dict, either the Cachito request ID or a dict
+                        with 'id' key
+
+        :return: str, the ICM as JSON
+        """
+        request_id = self._get_request_id(request)
+        url = '{}/{}/content-manifest'.format(self.api_endpoint, request_id)
+        logger.debug('Making request %s', url)
+        response = self.session.get(url)
+        response.raise_for_status()
+        icm = json.dumps(response.json())
+        logger.debug('Cachito ICM request response:\n%s', icm)
+        # Validate
+        read_yaml(icm, 'schemas/content_manifest.json')
+        return icm
```

Comment:
Since we are still going to manipulate the ICM later, should we return the dict here instead? So we won't need to decode it again later.

Reply:
Aha. Major question which @rcerven and I discussed. I wasn't certain if the ICM would be further mangled or not :)